pkg/kn/flags/resources_test.go

```diff
@@ -0,0 +1,86 @@
+// Copyright © 2020 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package flags
+
+import (
+	"testing"
+
+	"gotest.tools/assert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+type resourceOptionsTestCase struct {
+	requests         string
+	limits           string
+	expectedRequests corev1.ResourceList
+	expectedLimits   corev1.ResourceList
+	expectedErr      bool
+}
+
+func parseQuantity(value string) resource.Quantity {
+	q, _ := resource.ParseQuantity(value)
+	return q
+}
+
+func TestResourceOptions(t *testing.T) {
+	cases := []*resourceOptionsTestCase{
+		{"memory=200Mi,cpu=200m",
+			"memory=1024Mi,cpu=500m",
+			corev1.ResourceList{corev1.ResourceMemory: parseQuantity("200Mi"),
```

Comment:
I would also add other units to this test... especially to verify that the message to user is correct when units wrong. For instances, using GB or GByte or other similar units that one might use but that would not parse.

Reply:
The bad units error messages are relayed from imported libs to user and we're not processing that message in `kn`. We do a check though whether the errors are raised in `pkg/kn/flags/resources_test.go`.